tasks/copy_files.yml

```diff
@@ -3,7 +3,7 @@
   file:
     path: "{{ item.destination_path | dirname }}"
     state: directory
-    mode: 0750
+    mode: "{{ item.directory_mode | default('0750') }}"
```

Comment:
can you confirm that the default value being a string is fine? its going from 0750 to '0750' ... im sure its fine, but worth asking

Reply:
I am pretty sure there are molecule tests for the `copy_files` feature. Would it be possible to extend those to verify this?

Reply:
Good idea, in the `plaintext-rhel` scenario, we test the copy file features. within `roles/confluent.test/molecule/plaintext-rhel/molecule.yml` we could have:
```
        zookeeper_copy_files:
          - source_path: "roles/confluent.test/molecule/{{scenario_name}}/molecule.yml"
            destination_path: /tmp/molecule.yml
          - source_path: "roles/confluent.test/molecule/{{scenario_name}}/molecule.yml"
            destination_path: /tmp/test/molecule2.yml
            directory_mode: 0700
            file_mode: 0600
```

and then under `roles/confluent.test/molecule/plaintext-rhel/verify.yml` you could add assertion tasks around the new file:
```
    - name: Get stats on second copied file
      stat:
        path: /tmp/test/molecule2.yml
      register: copied_file2

    - name: Assert second copied file's permissions correct
      assert:
        that:
          - copied_file2.stat.exists
          - copied_file2.stat.gr_name == 'confluent'
          - copied_file2.stat.pw_name == 'cp-kafka'
          # insert assertion on the mode here, should be copied_file2.stat.something
        quiet: true
```

Reply:
Hi @domenicbove string value for file/directory mode is allowed as per [ansible documentation](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/copy_module.html#parameter-mode). I will add a test in `plaintext-rhel` scenario